File: frontend/src/lib/api/sheets.ts
import { apiFetch } from "@/lib/api-client";

export type SheetConnectionStatus = {
  name: string;
  sheet_id: string;
  connected: boolean;
  worksheets: string[];
  error: string | null;
};

export type SheetsStatusResponse = {
  sheets: SheetConnectionStatus[];
};

export function getSheetsStatus(): Promise<SheetsStatusResponse> {
  return apiFetch<SheetsStatusResponse>("/sheets/status");
}

export type OrphanEntry = {
  link_ref_code: string;
  present_in: string[];
  missing_from: string[];
};

export type DestinationOrphanReport = {
  destination: string;
  tabs_checked: string[];
  orphans: OrphanEntry[];
};

export type OrphanCheckResponse = {
  reports: DestinationOrphanReport[];
};

export function getOrphanReport(): Promise<OrphanCheckResponse> {
  return apiFetch<OrphanCheckResponse>("/sheets/orphans");
}

File: backend/app/schemas/sheets.py
from pydantic import BaseModel


class SheetConnectionStatus(BaseModel):
    name: str
    sheet_id: str
    connected: bool
    worksheets: list[str] = []
    error: str | None = None


class SheetsStatusResponse(BaseModel):
    sheets: list[SheetConnectionStatus]


class OrphanEntry(BaseModel):
    link_ref_code: str
    present_in: list[str]
    missing_from: list[str]


class DestinationOrphanReport(BaseModel):
    destination: str
    tabs_checked: list[str]
    orphans: list[OrphanEntry]


class OrphanCheckResponse(BaseModel):
    reports: list[DestinationOrphanReport]
